snap/info.go

```diff
@@ -898,6 +898,10 @@ type PlugInfo struct {
 	Label     string
 	Apps      map[string]*AppInfo
 	Hooks     map[string]*HookInfo
+
+	// ComponentHooks is a mapping of component names to a mapping of component
+	// hook names to information about the hook.
+	ComponentHooks map[string]map[string]*HookInfo
```

Comment:
I'm not sure that binding Plugs to component hooks in the context of the snap itself make sense because of implicit hooks that will need a different approach, we need to generate the corresponding profiles possibly independently of the snap in general. This is going to be a complex part of the problem, more than the running of the hook themselves.